pkg/render/kubecontrollers/kube-controllers.go

```diff
@@ -290,57 +352,40 @@ func (c *kubeControllersComponent) controllersRole() *rbacv1.ClusterRole {
 			APIGroups:     []string{"policy"},
 			Resources:     []string{"podsecuritypolicies"},
 			Verbs:         []string{"use"},
-			ResourceNames: []string{"calico-kube-controllers"},
+			ResourceNames: []string{c.kubeControllerName},
 		})
 	}
 
 	return role
 }
 
-func (c *kubeControllersComponent) controllersRoleBinding() *rbacv1.ClusterRoleBinding {
-	return &rbacv1.ClusterRoleBinding{
-		TypeMeta: metav1.TypeMeta{Kind: "ClusterRoleBinding", APIVersion: "rbac.authorization.k8s.io/v1"},
-		ObjectMeta: metav1.ObjectMeta{
-			Name:   "calico-kube-controllers",
-			Labels: map[string]string{},
-		},
-		RoleRef: rbacv1.RoleRef{
-			APIGroup: "rbac.authorization.k8s.io",
-			Kind:     "ClusterRole",
-			Name:     "calico-kube-controllers",
-		},
-		Subjects: []rbacv1.Subject{
-			{
-				Kind:      "ServiceAccount",
-				Name:      "calico-kube-controllers",
-				Namespace: common.CalicoNamespace,
-			},
-		},
-	}
-}
-
 func (c *kubeControllersComponent) controllersDeployment() *appsv1.Deployment {
 	env := []corev1.EnvVar{
+		{Name: "KUBE_CONTROLLERS_CONFIG_NAME", Value: c.kubeControllerConfigName},
 		{Name: "DATASTORE_TYPE", Value: "kubernetes"},
 	}
 
 	env = append(env, c.cfg.K8sServiceEp.EnvVars(false, c.cfg.Installation.KubernetesProvider)...)
 
-	enabledControllers := []string{"node"}
-	if c.cfg.Installation.Variant == operatorv1.TigeraSecureEnterprise {
-		enabledControllers = append(enabledControllers, "service", "federatedservices")
+	enabledControllers := make([]string, 0)
+	if c.kubeControllerName == KubeController {
+		enabledControllers = append(enabledControllers, "node")
```

Comment:
I think I'd like to see the decision of what controllers are enabled set in the constructors. Then you can render set them here.